### PR TITLE
mzexplore,optbench: minor fixes

### PR DESCRIPTION
**Pull request by aalexandrov**
It turns out that [pg8000.native.identifier](https://github.com/tlocke/pg8000/blob/017959e97751c35a3d58bc8bd5722cee5c10b656/pg8000/converters.py#L739-L761) is not compatible with [the Postgres code](https://github.com/postgres/postgres/blob/b0f7dd915bca6243f3daf52a81b8d0682a38ee3b/src/backend/utils/adt/ruleutils.c#L11968-L12050
), so we need to roll our own modified `identifier` version.

I'm using the occasion to also make the `mzexplore` code a bit more robust and fix a warning in `optbench`.

### Motivation

  * This PR fixes a previously unreported bug.

### Tips for reviewer

In `mzexplore`:
- Fix `identifier` quotation.
- Make `items.sql` query case-insensitive.
- Skip failing `CREATE` or `EXPLAIN` commands with warning.

In `optbench`:
- Adress pandas deprecation warning.

### Checklist

- [ ] This PR has adequate test coverage / QA involvement has been duly considered.
- [ ] This PR has an associated up-to-date [design doc](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/design/README.md), is a design doc ([template](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/design/00000000_template.md)), or is sufficiently small to not require a design.
  
- [ ] If this PR evolves [an existing `$T ⇔ Proto$T` mapping](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/command-and-response-binary-encoding.md) (possibly in a backwards-incompatible way), then it is tagged with a `T-proto` label.
- [ ] If this PR will require changes to cloud orchestration or tests, there is a companion cloud PR to account for those changes that is tagged with the release-blocker label ([example](https://github.com/MaterializeInc/cloud/pull/5021)).
  
- [ ] This PR includes the following [user-facing behavior changes](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/guide-changes.md#what-changes-require-a-release-note):
  - There are no user-facing behavior changes.

### CODE REVIEW

File: misc/python/materialize/mzexplore/sql.py [hunk 1]
```diff
@@ -117,3 +118,41 @@ def try_mzfmt(sql: str) -> str:
...
+    if not isinstance(sql, str):
+        raise InterfaceError("identifier must be a str")
+
+    if len(sql) == 0:
+        raise InterfaceError("identifier must be > 0 characters in length")
+
+    def is_alpha(c: str) -> bool:
+        return ord(c) >= ord("a") and ord(c) <= ord("z") or c == "_"
+
+    def is_alpanum(c: str) -> bool:
```
Comment (benesch) [inline misc/python/materialize/mzexplore/sql.py:140]:
```suggestion
    def is_alphanum(c: str) -> bool:
```

File: misc/python/materialize/mzexplore/sql.py [hunk 2]
```diff
@@ -117,3 +118,41 @@ def try_mzfmt(sql: str) -> str:
...
+        return is_alpha(c) or ord(c) >= ord("0") and ord(c) <= ord("9")
+
+    quote = not (is_alpha(sql[0]))
```
Comment (benesch) [inline misc/python/materialize/mzexplore/sql.py:143]: Note this won't be sufficient in all cases. An identifier composed of all alphanumeric characters may need quoting if it is a reserved keyword (e.g., `FROM`). Materialize is pretty permissive about this, but there are a few cases where we _have_ to treat reserved keywords as keywords and not identifiers if unquoted.

**Review by benesch — APPROVED** [review]
What a bad bug in pg8000. 🤦🏽‍♂️

**Review by vmarcos — APPROVED** [review]
Thank you!

File: misc/python/materialize/mzexplore/sql.py (hunk 2 above)
Comment (aalexandrov) [reply misc/python/materialize/mzexplore/sql.py]: Good catch, I guess it was too late for me to double-check this. I added the list of keywords as a soft link under the `sql` folder and an extra check that the identifier string `s` is not a keyword

```python
if s.upper() in keywords():
    quote = True
```
where
```python
@functools.lru_cache(maxsize=1)
def keywords() -> set[str]:
    """
    Return a list of keywords reserved by Materialize.
    """
    with resource_path("sql/keywords.txt").open() as f:
        return set(
            line.strip().upper()
            for line in f.readlines()
            if not line.startswith("#") and len(line.strip()) > 0
        )
```

**Pull request merged**